ts/sagas/pinlogin.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * A saga that manages the Onboarding.
+ *
+ * For a detailed view of the flow check @https://docs.google.com/document/d/1le-IdjcGWtmfrMzh6d_qTwsnhVNCExbCd6Pt4gX7VGo/edit
+ */
+
+import { NavigationActions } from "react-navigation";
+import { call, Effect, put, takeLatest } from "redux-saga/effects";
+
+import ROUTES from "../navigation/routes";
+import {
+  PIN_LOGIN_INITIALIZE,
+  PIN_LOGIN_VALIDATE_FAILURE,
+  PIN_LOGIN_VALIDATE_REQUEST
+} from "../store/actions/constants";
+import { PinValidateRequest } from "../store/actions/pinlogin";
+import { getPin } from "../utils/keychain";
+
+/**
+ * The PIN step of the pin login
+ */
+function* pinLoginSaga(): Iterator<Effect> {
+  const navigateToPinLoginNavigatorAction = NavigationActions.navigate({
+    routeName: ROUTES.PINLOGIN,
+    key: undefined
```

Comment:
what does `key: undefined` do?

Reply:
key is an optional parameter string or null,  if set, navigation will go back from the given key. If null, navigation will go back anywhere.